api/v1/apiserver_types.go

```diff
@@ -185,6 +198,25 @@ type APIServerDeploymentSpec struct {
 	// Template describes the API server Deployment pod that will be created.
 	// +optional
 	Template *APIServerDeploymentPodTemplateSpec `json:"template,omitempty"`
+
+	// The deployment strategy to use to replace existing pods with new ones.
+	// +optional
+	// +patchStrategy=retainKeys
+	Strategy *APIServerDeploymentStrategy `json:"strategy,omitempty" patchStrategy:"retainKeys" protobuf:"bytes,4,opt,name=strategy"`
```

Comment:
I'd want to hear what the use case is that we need to add this override for the apiserver. In other words why would someone need to change the Strategy?

Reply:
you are right, no need to allow overriding the strategy. But it would be better to set it to RollingUpdate instead of Recreate to increase HA during an update.

Reply:
@tmjd I think the `RollingUpdate` is better because it replaces pods gradually to ensure minimal downtime and new pods are created before old ones are terminated. The `Recreate` just deletes all existing pods simultaneously and then creates new ones, causing downtime.

Reply:
Yeah, I think RollingUpdate is preferable most of the time. There are some subset of scenarios where Recreate is necessary - namely for 1 node clusters running on EKS, or similar platforms, that run our API server in host network.

Can we configure RollingUpdate when hostNet: false, and Recreate otherwise? would that work?

Reply:
Yes, configured to use RollingUpdate when `hostNetwork: false`. thanks for the suggestion.